ts/features/it-wallet/screens/issuing/ItwPidRequestScreen.tsx

```diff
@@ -0,0 +1,129 @@
+import { InfoScreenComponent } from "../../../fci/components/InfoScreenComponent";
+import { ItWalletError } from "../../utils/errors/itwErrors";
+import { mapRequirementsError } from "../../utils/errors/itwErrorsMapping";
+import { Pictogram } from "../../../../components/core/pictograms";
+import ItwLoadingSpinnerOverlay from "../../components/ItwLoadingSpinnerOverlay";
+import { itwActivationStop } from "../../store/actions";
+import { ITW_ROUTES } from "../../navigation/routes";
+
+/**
+ * ItwPidRequestScreen's navigation params.
+ * The pidData consists of the data needed to request a PID.
+ */
+export type ItwPidRequestScreenNavigationParams = {
+  pidData: PidData;
+};
+
+/**
+ * Renders a preview screen which requests a PID.
+ */
+const ItwPidRequestScreen = () => {
+  const route =
+    useRoute<
+      Route<"ITW_ACTIVATION_PID_REQUEST", ItwPidRequestScreenNavigationParams>
+    >();
+  const navigation = useNavigation<IOStackNavigationProp<ItwParamsList>>();
+  const dispatch = useIODispatch();
+  const pid = useIOSelector(itwPidSelector);
+
+  /**
+   * Dispatches the PID request on first render.
+   */
+  useOnFirstRender(() => {
+    dispatch(itwPid.request(route.params.pidData));
+  });
+
+  /**
+   * Navigates to the PID preview screen when the PID is available.
+   * This prevents the "cannot update a component while rendering a different component" error
+   * if the navigation is called directly during the pot fold.
+   */
+  useEffect(() => {
+    if (pot.isSome(pid)) {
+      navigation.navigate(ITW_ROUTES.ACTIVATION.PID_PREVIEW);
+    }
+  }, [navigation, pid]);
+
+  /**
+   * Renders the error view.
+   */
+  const ErrorView = (error: ItWalletError) => {
+    const mappedError = mapRequirementsError(error);
+    const cancelButtonProps = {
+      block: true,
+      light: false,
+      bordered: true,
+      onPress: () => dispatch(itwActivationStop()),
+      title: I18n.t("features.itWallet.generic.close")
+    };
+    return (
+      <>
+        <InfoScreenComponent
+          title={mappedError.title}
+          body={mappedError.body}
+          image={<Pictogram name="error" />}
+        />
+        <FooterWithButtons
+          type={"SingleButton"}
+          leftButton={cancelButtonProps}
+        />
+      </>
+    );
+  };
+
+  /**
+   * Renders the loading spinner.
+   * @returns a loading spinner overlay
+   */
+  const LoadingView = () => (
+    <ItwLoadingSpinnerOverlay
+      captionTitle={I18n.t("features.itWallet.issuing.loading.title")}
+      isLoading
+      captionSubtitle={I18n.t("features.itWallet.issuing.loading.subtitle")}
+    >
+      <></>
+    </ItwLoadingSpinnerOverlay>
+  );
+
+  const RenderMask = () =>
+    pot.fold(
+      pid,
+      () => <LoadingView />,
+      () => <LoadingView />,
+      () => <LoadingView />,
+      err => ErrorView(err),
+      () => <LoadingView />,
+      () => <LoadingView />,
+      () => <LoadingView />,
+      (_, err) => ErrorView(err)
+    );
```

Comment:
I don't know if it could be better use a `if (pot.isLoading(pid) || ....) {return <LoadingView />;}`. Maybe pot could be overkilled here. This is a personal consideration.

Reply:
Personally, I prefer the idiomatic functional way of folding the pot even if in this case it definitely _is_ overkill.